trollflow2/launcher.py

```diff
@@ -438,3 +440,61 @@ def sendmail(config, trace):
     pid = Popen([sendmail, "-t", "-oi"], stdin=PIPE)
     pid.communicate(msg.as_bytes())
     pid.terminate()
+
+
+def launch(args_in):
+    """Launch the processing."""
+    args = parse_args(args_in)
+
+    log_config = args.pop("log_config", None)
+    if log_config is not None:
+        with open(log_config) as fd:
+            log_config = yaml.safe_load(fd.read())
+
+    logger = logging.getLogger("satpy_launcher")
+
+    log_queue = Manager().Queue()
+
+    with logging_on(log_queue, log_config):
+        logger.warning("Launching Satpy-based runner.")
+        product_list = args.pop("product_list")
+        test_message = args.pop("test_message")
+        threaded = args.pop("threaded")
+        connection_parameters = args
+
+        runner = Runner(product_list, log_queue, connection_parameters, test_message, threaded)
+        runner.run()
+
+
+def parse_args(args_in):
+    """Parse commandline arguments."""
+    parser = argparse.ArgumentParser(
+        description='Launch trollflow2 processing with Satpy listening on the specified Posttroll topic(s)')
+    parser.add_argument("topic", nargs='*',
+                        help="Topic to listen to",
+                        type=str)
+    parser.add_argument("product_list",
+                        help="The yaml file with the product list",
+                        type=str)
+    parser.add_argument("-m", "--test_message",
+                        help="File path with the message used for testing offline. This implies threaded running.",
+                        type=str, required=False)
+    parser.add_argument("-t", "--threaded",
+                        help="Run the product generation in threads instead of processes.",
+                        action='store_true')
+
+    parser.add_argument("-c", "--log-config",
+                        help="Log config file (yaml) to use",
+                        type=str, required=False)
+    parser.add_argument('-n', "--nameserver", required=False, type=str,
+                        help="Nameserver to connect to", default='localhost')
```

Comment:
In the main documentation, you write that we can pass `-n False`.  This should be documented here as well.

Would that be for use with a test message?  Otherwise, how would trollflow2 get any input?

Reply:
Argument parser help message updated in https://github.com/pytroll/trollflow2/pull/159/commits/3c5b38605c995c95a0c90d3e97cf17a79d5b422e

If the `nameserver` is disabled, the messages are received via direct port connection to source defined with `-a` argument as shown in the documentation.